src/Language/PureScript/TypeChecker/Monad.hs

```diff
@@ -200,16 +201,17 @@ modifyEnv f = modify (\s -> s { checkEnv = f (checkEnv s) })
 -- |
 -- Run a computation in the Check monad, starting with an empty @Environment@
 --
-runCheck :: (MonadError MultipleErrors m) => Check a -> m (a, Environment)
+runCheck :: (MonadError MultipleErrors m, MonadWriter MultipleErrors m) => Check a -> m (a, Environment, MultipleErrors)
 runCheck = runCheck' initEnvironment
 
 -- |
 -- Run a computation in the Check monad, failing with an error, or succeeding with a return value and the final @Environment@.
 --
-runCheck' :: (MonadError MultipleErrors m) => Environment -> Check a -> m (a, Environment)
-runCheck' env c = interpretMultipleErrors $ do
-  (a, s) <- flip runStateT (CheckState env 0 0 Nothing) $ unCheck c
-  return (a, checkEnv s)
+runCheck' :: (MonadError MultipleErrors m, MonadWriter MultipleErrors m) => Environment -> Check a -> m (a, Environment, MultipleErrors)
```

Comment:
Yes, you are right. I'm not using the constraint and also running `Writer` twice. I fixed both issues.
I returned `MultipleErrors` to use `tell` outside inside `compile'`, as we can't use `tell` in `runCheck`.
Do you think that is a bad decission?
Thanks!